### PR TITLE
Replace ASCII escape sequences with unicode symbols

### LINKED ISSUE #486: Replace ASCII escapes with fancy Unicode symbols
I find the `%[` things hard to read in source code. Using Unicode characters instead might be more readable. That would bring us here:

 * `⇥` increases the indentation level.
 * `⇤` decreases the indentation level.
 * `«` begins a statement. For multiline statements, every line after the first line is double-indented.
 * `»` ends a statement.

### PR DESCRIPTION
**Pull request by cwwwu**
Closes #486

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt [hunk 1]
```diff
@@ -194,6 +194,12 @@ class CodeBlock private constructor(
       }
 
       while (p < format.length) {
+        if (format[p] in setOf('⇥', '⇤', '«', '»')) {
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: How about checking for these symbols in `isNoArgPlaceholder()`?

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt [hunk 2]
```diff
@@ -251,8 +257,14 @@ class CodeBlock private constructor(
...
+          formatParts += format[p].toString()
+          p++
+          continue
+        }
+
         if (format[p] != '%') {
-          var nextP = format.indexOf('%', p + 1)
+          var nextP = format.indexOfAny(listOf("%", "«", "»", "⇥", "⇤"), p + 1)
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: nit: Use the version that takes a `CharArray`

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt (hunk 1 above)
Comment (cwwwu) [reply src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: There are a handful of places that use `isNoArgPlaceholder()` assuming that it's prefixed with a `%`. I'll rework this a bit (maybe a new method to check for single character argument placeholders)

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt (hunk 2 above)
Comment (cwwwu) [reply src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: 👍Fixed that. I also found a bug in the midst of making that change in handling the new single character placeholders in `addNamed` 😄

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt [hunk 3]
```diff
@@ -90,7 +90,7 @@ class CodeBlock private constructor(
       }
 
       // If the matching format part has an argument, check that too.
-      if (formatPart.startsWith("%") && !isNoArgPlaceholder(formatPart[1])) {
+      if (formatPart.startsWith("%") && !isMultiCharNoArgPlaceholder(formatPart[1])) {
```
Comment (Egorand) [inline src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: Nice, I like the name.

File: src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt [hunk 4]
```diff
@@ -107,8 +83,8 @@ class CodeBlockTest {
     val map = LinkedHashMap<String, Any>()
     map.put("text", "tacos")
     val block = CodeBlock.builder()
-        .addNamed("%>\n%text:L for %%3.50", map).build()
-    assertThat(block.toString()).isEqualTo("\n    tacos for %3.50")
+        .addNamed("⇥\n%text:L for\n⇤%%3.50", map).build()
```
Comment (Egorand) [inline src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt]: Any specific reason for this test change? Does the code still work with the old test case?

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt (hunk 2 above)
Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: Ha, nice. Any new test cases that we should add?

File: src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt (hunk 4 above)
Comment (cwwwu) [reply src/test/java/com/squareup/kotlinpoet/CodeBlockTest.kt]: Yes, it still works. I changed this test case as a result of encountering that bug I mentioned above. The key difference here is that there is a single no-arg placeholder in the middle of code block.

File: src/main/java/com/squareup/kotlinpoet/CodeBlock.kt (hunk 2 above)
Comment (cwwwu) [reply src/main/java/com/squareup/kotlinpoet/CodeBlock.kt]: I ended up modifying the test `namedAndNoArgFormat` for this.

**Pull request merged**